src/wp-includes/block-template-utils.php

```diff
@@ -1177,6 +1185,16 @@ function get_block_templates( $query = array(), $template_type = 'wp_template' )
 			) {
 				$query_result[] = _build_block_template_result_from_file( $template_file, $template_type );
 			}
+
+			// The custom templates with no associated post types are available for all post types.
+			if ( isset( $query['post_type'] ) && ! isset( $template_file['postTypes'] ) ) {
+				$candidate              = _build_block_template_result_from_file( $template_file, $template_type );
+				$default_template_types = get_default_block_template_types();
+				$is_custom              = ! isset( $default_template_types[ $candidate->slug ] );
+				if ( $is_custom ) {
```

Comment:
```suggestion
				if ( ! isset( $default_template_types[ $candidate->slug ] ) ) {
```

This seems the more logical attempt here?

Reply:
Sounds good to me. Thanks for your suggestion. :+1: